### DISCUSSION TITLE
Um componente que poderia fazer mais de uma requisição

### DISCUSSION THREAD

**Discussion by mister-vini** [frontendbr/forum#1035] [Perguntas e Respostas]
Boa noite, estou tendo um problema e quero fazer uma função que eu reutilizaria em outras ações para fazer requisição, estou passando um método que é um objeto de filtros e queria utilizar, mas estou tendo problemas.
Obrigado pela atenção

**Comment by felipefialho** [comment]
@mister-vini Especifica melhor os problemas, de preferencia colocando exemplos de códigos :)

**Comment by mister-vini** [comment]
![capturar_certa](https://user-images.githubusercontent.com/38114375/38837725-6f9ef6c6-41a9-11e8-951f-38f2ed9a478e.PNG)
![capturar](https://user-images.githubusercontent.com/38114375/38837726-6fc107fc-41a9-11e8-96d8-cf0bca328369.PNG)
Eu tenho um componente que faz a requisição, mas gostaria de fazer mais de uma requisição e transformar ele em reutilizável, como poderia fazer isso, eu tenho outro componente com os parâmetros que preciso chamar

**Comment by mister-vini** [comment]
Obrigado @LFeh e desculpa alguma coisa

**Comment by ghost** [comment]
@mister-vini se é bem o que entendi, talvez possa te ajudar rxjs com observable ou redux-saga com generators e saga.

Sobre Observable:
[Observable](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html)

Aqui um artigo sobre generators:
[Generators](https://braziljs.org/blog/generators-yield-e-iterators/)

Isso vai resolver bem o lance de mais requisições de forma assíncronas. 

Sobre reutilizar, você pode deixar essas funções como filterBy externas e puras pra que tu possa passar parâmetros e possa reaproveitar.

Não sei se essas explicações são muito abstratas mas um caminho é ver esses dois padrões que te falei acima (Saga e Observable).

**Comment by carlosqsilva** [comment]
Se a próxima requisição for independente da primeira, acredito que você possa usar:
```js
Promise.all([request, request])
 .then(results => {
    const r1 = results[0]
    const r2 = results[1]
.....
})
```

**Comment by ninetails** [comment]
Assumindo que pela estrutura c está utilizando `redux-thunk`, foi bem o que o meu xará falou acima.

Se quiser algo mais explicado pelo próprio Dan Abramov: https://github.com/reactjs/redux/issues/723#issuecomment-139927639

**Comment by torto** [comment]
Você pode usar `Promisse.all` que foi citado acima, ou se você precisa do resultado de uma requisição para fazer a outra, você pode usar async await que você já está usando... porém refatoraria esse cód. para algo assim:
```
return async function(dispacth) {
  dispacth({
      type: load,
      payload: {featching: true}
    })
  try {
    const response1 = await axios.get(URL);
    const response2 = await axios.get(`url?test=${response1.algumacoisa}`)
    return dispacth({
      type: sucess,
      payload: {response1, response2}
    })
  } catch(e){
    return dispacth({
      type: error,
      payload: {e}
    })
  }
}
```
Sobre usar Observable, generator e yields da vida eu acho muito complicado e de difícil manutenção, o simples é sempre a melhor opção :heart: :dancing_women:

**Comment by mister-vini** [comment]
Obrigado Pessoal me ajudou muito! Ja foi resolvido, obrigado mesmo!
